src/transforms/remap.rs

```diff
@@ -183,50 +187,60 @@ impl TransformConfig for RemapConfig {
         Input::all()
     }
 
-    fn outputs(&self, merged_definition: &schema::Definition) -> Vec<Output> {
+    fn outputs(&self, input_definition: &schema::Definition) -> Vec<Output> {
         // We need to compile the VRL program in order to know the schema definition output of this
         // transform. We ignore any compilation errors, as those are caught by the transform build
         // step.
         let default_definition = self
             .compile_vrl_program(
                 enrichment::TableRegistry::default(),
-                merged_definition.clone(),
+                input_definition.clone(),
             )
-            .ok()
-            .and_then(|(_, _, _, state)| {
+            .map(|(_, _, _, state)| {
                 let meaning = state
                     .get_external_context::<MeaningList>()
                     .cloned()
                     .expect("context exists")
                     .0;
 
-                state
-                    .target_kind()
-                    .clone()
-                    .into_object()
-                    .map(schema::Definition::from)
-                    .map(|mut def| {
-                        for (id, path) in meaning {
-                            def = def.with_known_meaning(path, &id);
-                        }
-                        def
-                    })
+                let mut new_type_def = Definition::empty_with_kind(
+                    state.target_kind().clone(),
+                    input_definition.log_namespaces().clone(),
+                );
+                for (id, path) in meaning {
+                    new_type_def = new_type_def.with_meaning(path, &id);
+                }
+                new_type_def
             })
-            .unwrap_or_else(schema::Definition::empty);
+            .unwrap_or_else(|_| {
+                Definition::empty_with_kind(
+                    Kind::never(),
+                    input_definition.log_namespaces().clone(),
+                )
+            });
 
         // When a message is dropped and re-routed, we keep the original event, but also annotate
         // it with additional metadata.
-        let dropped_definition = merged_definition.clone().with_field(
-            log_schema().metadata_key(),
-            Kind::object(BTreeMap::from([
-                ("reason".into(), Kind::bytes()),
-                ("message".into(), Kind::bytes()),
-                ("component_id".into(), Kind::bytes()),
-                ("component_type".into(), Kind::bytes()),
-                ("component_kind".into(), Kind::bytes()),
-            ])),
-            Some("metadata"),
-        );
+        let mut dropped_definition =
+            Definition::empty_with_kind(Kind::never(), input_definition.log_namespaces().clone());
+
+        // The vector namespace appends the dropped fields to the "event metadata", which doesn't yet have a schema
```

Comment:
Is that something to be resolved in this PR?

Reply:
No. I'm following up with an additional PR to add type information to metadata which will implement this.